### PR TITLE
RavenDB-16950 Added CompareValue case for two LazyNumber using AlmostEquals instead string comparison.

### PR DESCRIPTION
**Pull request by maciejaszyk**

### CODE REVIEW

File: src/Raven.Client/Json/BlittableOperation.cs [hunk 1]
```diff
@@ -223,6 +224,14 @@ private static bool CompareValues(BlittableJsonReaderObject.PropertyDetails oldP
                 return @double % 1 == 0 && @long.Equals((long)@double);
             }
 
+            if (oldProp.Token == BlittableJsonToken.LazyNumber && newProp.Token == BlittableJsonToken.LazyNumber)
+            {
```
Comment (ayende) [inline src/Raven.Client/Json/BlittableOperation.cs:228]: Can we check the raw string value first, to avoid the allocations?

Comment (maciejaszyk) [reply src/Raven.Client/Json/BlittableOperation.cs:228]: I think it's already done by first condition in 102 line.  It's first comparing length of string and when new and old are equal run Memory.CompareInline method.

Comment (ayende) [reply src/Raven.Client/Json/BlittableOperation.cs:228]: Correct

**Review by ayende — APPROVED** [review]

File: test/SlowTests/Issues/RavenDB_16950.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        {
+        }
+
+        [Fact]
+        public void CheckIfDecimalValueNotChangesAfterComputation()
+        {
+            int changesCounter;
+            using (var store = GetDocumentStore())
+            {
+                store.Maintenance.Send(new CreateSampleDataOperation());
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB_16950.cs:20]: Let's avoid doing that if possible. Takes a while and you need only 1 entry in the test.

File: test/SlowTests/Issues/RavenDB_16950.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    Order order = session.Load<Order>("orders/823-A");
+                    order.ShipTo.Location.Latitude = order.ShipTo.Location.Latitude;
+                    var changes = session.Advanced.WhatChanged();
+                    changesCounter = changes.Count;
+                }
+            }
+            Assert.Equal(0, changesCounter);
+        }
+
+        public class Address
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB_16950.cs]: private

File: test/SlowTests/Issues/RavenDB_16950.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            public Location Location { get; set; }
+        }
+
+        public class Location
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB_16950.cs]: private

File: test/SlowTests/Issues/RavenDB_16950.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            public double Latitude { get; set; }
+            public double Longitude { get; set; }
+        }
+
+        public class Order
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB_16950.cs]: private

**Pull request merged**